### PR TITLE
Feature 1014774

### PR DESCRIPTION
**Pull request by MarieBecks**
#1034781: Improved error messages

### CODE REVIEW

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/ShowErrorUtil.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    String smallExceptionMessage = _identifyException(pE);
+
+    if (loadError != null)
+    {
+      smallExceptionMessage = _identifyException(loadError);
+      exceptionMessage += BootstrapUtil.stackTraceToString(loadError) + "\n\n";
+    }
+    exceptionMessage += BootstrapUtil.stackTraceToString(pE);
+
+    showError(smallExceptionMessage, exceptionMessage);
```
Comment (jboesl) [inline jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/ShowErrorUtil.java]: Bad code: you shouldn't actually do stuff in constructors like showing dialogs.

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/Starter.java [hunk 2]
```diff
@@ -37,7 +35,7 @@ public static void main(String[] args) throws Throwable
       _loadNewVersion(url);
       args[0] = url.toExternalForm();
     }
-    catch (Throwable pE)
+    catch (IOException pE)
```
Comment (jboesl) [inline jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/Starter.java]: Why do you catch solely IOExceptions here? '_loadNewVersion(url)' might throw RuntimeExceptions or maybe even errors.

File: jloadr/src/main/java/de/adito/jloadr/Main.java [hunk 3]
```diff
@@ -66,6 +67,7 @@ public static void run(String pUrl, String pIconPath, String pStartName) throws
           loaderConfig.load(inputStream);
         }
 
+        //client wird ausgeführt
```
Comment (jboesl) [inline jloadr/src/main/java/de/adito/jloadr/Main.java]: This should be english.

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/ShowErrorUtil.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
+package de.adito.jloadr.bootstrap;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.*;
+import java.net.*;
+
```
Comment (jboesl) [inline jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/ShowErrorUtil.java:7]: Author is missing.

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/Starter.java [hunk 5]
```diff
@@ -47,11 +45,7 @@ public static void main(String[] args) throws Throwable
     }
     catch (Throwable pE)
     {
-      String exceptionMessage = "";
-      if (loadError != null)
-        exceptionMessage += BootstrapUtil.stackTraceToString(loadError) + "\n\n";
-      exceptionMessage += BootstrapUtil.stackTraceToString(pE);
-      _showError(exceptionMessage);
+      new ShowErrorUtil(loadError, pE);
```
Comment (jboesl) [inline jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/Starter.java]: Creating an object that does stuff solely upon creation is bad code.

**Review by jboesl — CHANGES_REQUESTED** [review]
I didn't check the new dialog and it's messages. I guess you did test that stuff. Take a look at my remarks. These should be easy to address.

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/DigestingInputStream.java [hunk 6]
```diff
@@ -50,7 +50,7 @@ public static MessageDigest getMessageDigest()
     }
     catch (NoSuchAlgorithmException pE)
     {
-      throw new RuntimeException();
+      throw new RuntimeException("Was not able to digest message.");
```
Comment (wglanzer) [inline jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/DigestingInputStream.java]: Do you really want to omit the stacktrace of "pE" here?

**Review by wglanzer — CHANGES_REQUESTED** [review]

File: jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/DigestingInputStream.java (hunk 6 above)
Comment (jboesl) [reply jloadr-bootstrap/src/main/java/de/adito/jloadr/bootstrap/DigestingInputStream.java]: Oops. Good catch.

**Review by jboesl — APPROVED** [review]

**Review by wglanzer — APPROVED** [review]

**Pull request merged**